homeassistant/components/media_player/toshiba.py

```diff
@@ -0,0 +1,220 @@
+"""
+Support for interface with a TOSHIBA CAST TV.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.toshiba/
+"""
+import logging
+import requests
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    SUPPORT_SELECT_SOURCE,
+    SUPPORT_TURN_ON, SUPPORT_TURN_OFF,
+    SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_STEP, MediaPlayerDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, STATE_OFF, STATE_ON, STATE_UNKNOWN, CONF_PORT)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+ICON = 'mdi:television'
+DEFAULT_NAME = 'Toshiba Cast TV'
+DEFAULT_PORT = 4430
+
+SUPPORT_TOSHIBATV = SUPPORT_VOLUME_STEP | \
+                    SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_SET | \
+                    SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Toshiba Cast TV platform."""
+    name = config.get(CONF_NAME)
+    port = config.get(CONF_PORT)
+
+    if discovery_info:
+        _LOGGER.debug('%s', discovery_info)
+        host = discovery_info.get('host')
+        port = discovery_info.get('port')
+        add_devices([ToshibaCastTVDevice(name, host, port)])
+        return True
+
+    host = config.get(CONF_HOST)
+
+    try:
+        add_devices([ToshibaCastTVDevice(name, host, port)])
+        return True
+    except OSError:
+        return False
+
+
+class ToshibaCastTVDevice(MediaPlayerDevice):
+    """Representation of a Toshiba cast TV."""
+
+    def __init__(self, name, host, port):
+        """Initialize the Toshiba device."""
+        self._name = name
+        self._muted = False
+        self._playing = True
+        self._state = STATE_UNKNOWN
+        self._host = host
+        self._port = port
+        self._volume = 0
+        self._uniqueid = None
+        self._current_source = None
+        self._source_list = ['ANT_CABLE', 'CAST', 'HDMI_1',
+                             'HDMI_2', 'HDMI_3', 'USB', 'AV']
+        self.update()
+
+
+
+    def update(self):
+        """Retrieve the latest data."""
+        try:
+            data = requests.get('https://{}:{:d}/v2/remote/uniqueid'.format
```

Comment:
Is it necessary to fetch this unique id on each update? Or would it be possible to just fetch it on startup? That would also be a nice way to test if the given configuration works.